spec/models/activity_spec.rb

```diff
@@ -0,0 +1,59 @@
+require 'rails_helper'
+
+RSpec.describe Activity, type: :model do
+  it { is_expected.to belong_to(:actor).inverse_of(:activities).class_name(User.name) }
+
+  describe '.create' do
+    let!(:user) { create(:user, name: 'tester') }
+
+    subject(:activity) { Activity.create(actor: user, object: user, event: :tested) }
```

Comment:
You didn't see my diagram correctly. I said that users will be using `Activity.new(...).notify(user).notify(current_course)...`